play 2.1.1

helllamer opened the pull request:
Yes, it works, but pull request contains one issue, that i'm unable to resolve.

It looks like scalac 2.9.2 contains some crazy bug, that raises exception on correct sbt.PlayExceptions-2.1.1. So, 2.9.3 is used. And this is the issue.

If 2.9.3-build will be published somewhere, fake scala version, hardcoded dependency url or other workaround is needed because sbt-plugin is looking for 2.9.2. So, changes in README will be also needed.

// And it looks like play-framework 2.1.1 cannot be built using 2.9.2 too.

helllamer committed:
migration to play 2.1.1 (not compatible with scala_2.9.2)

knuton commented:
Thank you! We are currently focusing on our product launch, but will look into this soon.

peteygao commented:
Has this issue been resolved yet? Would really love to be able to use this Stylus plugin with Play 2.1.x!

helllamer commented:
Hi, Peter.

I think, this issue cannot be fixed until sbt-plugin uses scala-2.9.2.

To use this stylus with play-2.1.1, you should use "fake" dependency.

You need to **download sources and patch build.sbt**, build play-stylus against
scala-2.9.3:

``` bash
$ git clone git://github.com/helllamer/play-stylus.git play-stylus-git
$ cd play-stylus-git
$ sed -i build.sbt -e 's/2.9.2/2.9.3/'
$ sbt
> compile
```

Now, you need to publish your build somethere, where you have full access (corporate repo or local repo). Type the `publish` or `publish-local` command in opened sbt's console.

After publishing, you should go to published repo ( _remote server_ or _~/.ivy2/local_) and **substitute scala version** "by hands".

``` bash
$ ssh www@myrepo.mycompany.com
$ cd /var/ivy2repo
$ cd patience
$ ls -la
drwxr-xr-x  3 www-data www-data 4096 Apr 23 15:10 play-stylus_2.9.3_0.12
```

Looks like sbt version `_0.12` is appended here. It MAY be not appended. Anyway, its time to create all possible combinations:

```
$ ln -s play-stylus_2.9.3_0.12 play-stylus_2.9.2_0.12
$ ln -s play-stylus_2.9.3_0.12 play-stylus_2.9.2
```

**Fix dependency** into project/plugin.sbt on your _localhost_.

``` bash
$ cd ~/mywork/play/myproject1/project
$ cat plugins.sbt | grep play-stylus
addSbtPlugin("patience" % "play-stylus" % "0.1.3" from "http://myrepo.mycompany.com/ivy2/patience/play-stylus_2.9.2_0.12/0.1.3/play-stylus-0.1.3.jar")
```

If you used `publish-local`, you will use `file:///home/user/.ivy2/local/patience/...` URL.

Some steps of this manual may be reduntant or inprecise, because this quest has been get two months ago, and something may be forgotten.

vendethiel commented:
you can update scala to 2.10.1

vendethiel commented:
@helllamer Getting this with your version (that I changed a bit to publish locally) : 

![](http://i.imgur.com/dgajdBf.png)

helllamer commented:
Please install stylus into your system. 'stylus' executable must be available from $PATH.

Nami-Doc notifications@github.com пишет:

> @helllamer Getting this with your version (that I changed a bit to
> publish locally) : 
> 
> ![](http://i.imgur.com/dgajdBf.png)
> 
> ---
> 
> Reply to this email directly or view it on GitHub:
> https://github.com/patiencelabs/play-stylus/pull/1#issuecomment-21414565

vendethiel commented:
It is available from my %path% : 

```
Microsoft Windows [version 6.2.9200]
(c) 2012 Microsoft Corporation. Tous droits réservés.

>stylus -V
0.34.1
```

helllamer committed:
Migration to play 2.2.x.

helllamer committed:
play-2.2.x: fixes

helllamer committed:
scalaVersion := 2.10.2 (according to play 2.2.x migration guide).

helllamer commented:
I've added support for play-2.2.x.
play-2.1.x support dropped out (never worked as expected).
And it works ok (i've tested).
I've also already updated README file for new version.
@knuton Could you please make and publish new build? And merge all this patches.

vendethiel commented:
Good :).

vendethiel commented:
?

knuton commented:
Sorry for the noise, I am apparently bad at using trackpads. ;)

Will look into this.

danielberndt commented:
Verified PR and published to http://repo.patience.io/
Thanks a lot for your support!
